src/packages/piral-oidc/src/types.ts

```diff
@@ -95,6 +125,10 @@ export interface OidcClient {
    * Extends the headers of the provided request.
    */
   extendHeaders(req: OidcRequest): void;
+  /**
+   * The internal oidc-client library User Manager events.
+   */
+  events: UserManagerEvents;
```

Comment:
Exposing it so directly couples the lib fully to `oidc-client` and its changes. One potential abstraction would be to have `OidcClient extends EventEmitter`. All these events (e.g., `addAccessTokenExpiring`) would then be event names, instead of methods that accept a callback.

Reply:
I agree with decoupling, but I think `EventEmitter` is a node-specific interface. I'm not sure of the browser alternative. Trying to do `export interface OidcClient extends EventEmitter {` is an error Cannot find name 'EventEmitter'

Reply:
Looks like it is https://developer.mozilla.org/en-US/docs/Web/API/EventTarget

Reply:
Sorry was referring to interface `EventEmitter` from piral-base; but I realized this one is quite tightly coupled to being used from the API and not other objects.

`EventTarget` is like the parent class for many things (incl. Node, which is the parent class for Element etc.), i.e., since its not a DOM element I would not use `EventTarget`.

Thinking if we should redefine `on`, `off`, ... here or if there is a better option ...

Reply:
Did you have any further thoughts on this?

I think just writing our own "events.onSilentRenew" and "events.offSilentRenew" would be easy, and give us easy type safety, but I can see wanting to do an implementation of `on("silent-renew", (cb))` `off("silent-renew")` for consistency.

Reply:
Hm. So 2 thoughts:

1. We could introduce a `BaseEventEmitter<T>` in `piral-base`, which has `T` for defining the event map. Then `EventEmitter` would just be `BaseEventEmitter<PiralEventMap>`.
2. We could put the event emission not on the client, but rather on the API. I am not sure if we really have to listen via the `client` or if listening via the `instance` would be better. In the latter case, pilets could listen, too.

1) would essentially follow the current approach, just with a bit more convenience, while 2) would be a new approach with the downside that events are not dispatched / cannot be retrieved via the client alone. You'd need the Piral instance (or be in a pilet) for that.

Reply:
Honestly I was thinking of that too, and we probably **do** want to put the event emitting onto the api. Right now, we aren't using the events, but we probably will eventually for things like updating profile info.

I also think we could just split that functionality off into a separate ticket or leave it for later until someone actually needs it. YAGNI for now.

Reply:
So should we remove the `events` key for now?

Reply:
I think we should just leave it until someone needs it. I was adding it preemptively because it was easy, but now that we're talking about porting over EventEmitter and speccing out some new architecture, its probably not worth the effort right now.